internal/releasesources/release_sources_test.go

```diff
@@ -0,0 +1,1359 @@
+					Arch:    queryParameters["arch"][0],
+				}]; ok {
+					w.WriteHeader(http.StatusOK)
+					_, _ = w.Write([]byte(response))
+					return
+				}
+				w.WriteHeader(http.StatusNotFound)
+			}))
+			defer releasesServer.Close()
+
+			supportLevelServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusOK)
+				_, _ = w.Write([]byte(supportLevelResponse))
+			}))
+			defer supportLevelServer.Close()
+
+			type CoreOpenshiftVersion struct {
+				models.OpenshiftVersion
+				Channel          string
+				CpuArchitectures pq.StringArray `gorm:"type:text[]"`
+			}
+
+			expectedResult := []CoreOpenshiftVersion{
+				{
+					OpenshiftVersion: models.OpenshiftVersion{
+						DisplayName:  swag.String("4.10.1"),
+						SupportLevel: swag.String("end of life"),
+						Default:      false,
+					},
+					CpuArchitectures: []string{"x86_64"},
+					Channel:          "stable",
+				},
+				{
+					OpenshiftVersion: models.OpenshiftVersion{
+						DisplayName:  swag.String("4.12.1"),
+						SupportLevel: swag.String("maintenance"),
+						Default:      false,
+					},
+					CpuArchitectures: []string{"x86_64"},
+					Channel:          "stable",
+				},
+				{
+					OpenshiftVersion: models.OpenshiftVersion{
+						DisplayName:  swag.String("4.13.1"),
+						SupportLevel: swag.String("production"),
+						Default:      false,
+					},
+					CpuArchitectures: []string{"x86_64", "s390x"},
+					Channel:          "stable",
+				},
+				{
+					OpenshiftVersion: models.OpenshiftVersion{
+						DisplayName:  swag.String("4.13.17"),
+						SupportLevel: swag.String("production"),
+						Default:      false,
+					},
+					CpuArchitectures: []string{"x86_64"},
+					Channel:          "stable",
+				},
+				{
+					OpenshiftVersion: models.OpenshiftVersion{
+						DisplayName:  swag.String("4.13.19"),
+						SupportLevel: swag.String("production"),
+						Default:      false,
+					},
+					CpuArchitectures: []string{"s390x"},
+					Channel:          "stable",
+				},
+				{
+					OpenshiftVersion: models.OpenshiftVersion{
+						DisplayName:  swag.String("4.14.0"),
+						SupportLevel: swag.String("production"),
+						Default:      false,
+					},
+					CpuArchitectures: []string{"x86_64"},
+					Channel:          "stable",
+				},
+				{
+					OpenshiftVersion: models.OpenshiftVersion{
+						DisplayName:  swag.String("4.14.1"),
+						SupportLevel: swag.String("production"),
+						Default:      true,
+					},
+					CpuArchitectures: []string{"x86_64", "ppc64le"},
+					Channel:          "stable",
+				},
+				{
+					OpenshiftVersion: models.OpenshiftVersion{
+						DisplayName:  swag.String("4.15.0-ec.2"),
+						SupportLevel: swag.String("beta"),
+						Default:      false,
+					},
+					CpuArchitectures: []string{"x86_64"},
+					Channel:          "candidate",
+				},
+			}
+
+			err = syncReleaseImages(db, handler.ReleaseSources, OpenShiftReleasesAPIClient{BaseUrl: releasesServer.URL})
```

Comment:
It's best to test public interface and not private methods. Testing private methods make tests fragile